cmd/tink-worker/container/container_test.go

```diff
@@ -0,0 +1,398 @@
+package container
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/docker/docker/api/types"
+	containertypes "github.com/docker/docker/api/types/container"
+	networktypes "github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/google/go-cmp/cmp"
+	specs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+)
+
+type mockCli struct {
+	client.ContainerAPIClient
+	response containertypes.ContainerCreateCreatedBody
+	inspect  types.ContainerJSON
+}
+
+func (mcli *mockCli) ContainerCreate(_ context.Context, _ *containertypes.Config, _ *containertypes.HostConfig, _ *networktypes.NetworkingConfig, _ *specs.Platform, _ string) (containertypes.ContainerCreateCreatedBody, error) {
+	return mcli.response, nil
+}
+
+func (mcli *mockCli) ContainerInspect(_ context.Context, _ string) (types.ContainerJSON, error) {
+	return mcli.inspect, nil
+}
+
+func (mcli *mockCli) ContainerRemove(_ context.Context, _ string, _ types.ContainerRemoveOptions) error {
+	return nil
+}
+
+func (mcli *mockCli) ContainerStart(_ context.Context, _ string, _ types.ContainerStartOptions) error {
+	return nil
+}
+
+func (mcli *mockCli) ContainerWait(_ context.Context, id string, _ containertypes.WaitCondition) (<-chan containertypes.ContainerWaitOKBody, <-chan error) {
+	status := make(chan containertypes.ContainerWaitOKBody)
+	err := make(chan error)
+	if id == "" {
+		go func() {
+			err <- fmt.Errorf("Container wait failed, container id is invalid")
+		}()
+	} else {
+		go func() {
+			status <- containertypes.ContainerWaitOKBody{StatusCode: 0}
+		}()
+	}
+
+	return status, err
+}
+
+func Test_RemoveContainer(t *testing.T) {
```

Comment:
typically test function names should be in the format:
>`func TestXxx(*testing.T)`
where Xxx does not start with a lowercase letter. The function name serves to identify the test routine.

--[ref](https://pkg.go.dev/testing)